src/soc/ibm/power9/istep_14_3.c

```diff
@@ -2,37 +2,501 @@
 
 #include <cpu/power/istep_14.h>
 
-static void p9_pcie_config(void)
+#include <console/console.h>
+#include <cpu/power/scom.h>
+
+#include "pci.h"
+
+static uint64_t pec_addr(uint8_t pec, uint64_t addr)
+{
+	return addr + pec*0x400;
+}
+
+static void init_pecs(const uint8_t *iovalid_enable)
+{
+	enum {
+		P9N2_PEC_ADDREXTMASK_REG = 0x4010C05,
+		PEC_PBCQHWCFG_REG = 0x4010C00,
+		PEC_NESTTRC_REG = 0x4010C03,
+		PEC_PBAIBHWCFG_REG = 0xD010800,
+
+		/* powerbus.c has these too */
+		MBOX_SCRATCH_REG1 = 0x00050038,
+		MBOX_SCRATCH_REG6_GROUP_PUMP_MODE = (1 << 23),
```

Comment:
According to my notes this should be PPC_BIT(23) instead. Remember that order of bits in bitfields is undefined when it comes to endianness but Hostboot relies on it being MSB to LSB.